### ISSUE TITLE
no-unused: Allow rest parameters when destructuring object in function argument

### ISSUE THREAD

**Issue by corydeppen** [ajafff/tslint-consistent-codestyle#22]
Looking at the [no-unused doc](https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-unused.md#ignoring-uninteresting-parameters-and-variables), I didn't expect to run into this warning. Using `1.5.0`, the following is an example of where I see a warning.

```json
"no-unused": [
  true,
  "ignore-parameters"
]
```

```ts
// [tslint] Variable 'props' is unused. (no-unused)
const UsersList: React.SFC<Props> = ({ users, ...props }) => {};
```

**Comment by ajafff** [comment]
There are two thing to note here:
1. `"ignore-parameters"` only works for real parameters. Destructured parameters create local variables, that are not ignored.
2. The exception for properties in destructuring with rest does not include the rest itself.

To make my second point clearer:
```ts
let {a: _a, ...rest} = {a: 1, b: 2, c: 3};
console.log(rest);
```
`_a` is unused, but that's ok in here, because it's the only way to remove property `a` from the object in a type safe way.

On the other hand:
```ts
let {a, ..._rest} = {a: 1, b: 2, c: 3}; // error for unused variable _rest
console.log(a);
```
`_rest` is unused and results in an error. That's expected because you can simply remove it without changing the semantics of the program.


Let me know if that answers your question / solves your problem.

**Comment by corydeppen** [comment]
Sorry, I went back and found the `...props` variable wasn't being passed on to the child component like I thought. The rule did, in fact, report an unused variable. I found other similar usages of rest parameters that were being used and none were flagged as unused.
